packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -244,7 +244,7 @@ export type DecorateQueryProcedure<
         query: Query<
           inferProcedureInput<TProcedure>,
```

Comment:
the first parameter is the Output and the 3rd parameter is the Output after select(), so if the 3rd is wrong we'll also need to change the 1st

I'm scratching my head though because I'm surprised we got so far with it being wrong, but you do seem to be correct

Reply:
I think it's just a feature which isn't used very often.
I don't even actually use it, I was just playing around with the query invalidator and tried to invalidate queries based on the current query data (`{ predicate => query.state.data.includes(...) }`) and then found out that the query data was typed to the input of the query, not the output.

Reply:
If you can get the first parameter changed too then we'll get this in :)